### PR TITLE
Push Notification in Demo App

### PR DESCRIPTION
**Pull request by howawong**
1. Call registerDevice after user is logged in.
2. Implement UNUserNotificationCenterDelegate functions in AppDelegate
3. Change bundle ID and enable Push in project settings

Connects Skygeario/chat-SDK-iOS#118

### CODE REVIEW

**Comment by howawong** [comment]
image
The above screenshot shows the look and feel of a push notification message in foreground.
This is done via `completionHandler([.alert, .badge, .sound])`, so I don't think I need to add whisper to Podfile at this moment.

File: Swift Chat Demo 2/LoginRegistrationViewController.swift [hunk 1]
```diff
@@ -49,6 +50,13 @@ class LoginRegistrationViewController: UIViewController {
                 SVProgressHUD.dismiss(withDelay: 3)
                 return
             }
+            let center = UNUserNotificationCenter.current()
+            center.requestAuthorization(options:[.badge, .alert, .sound]) { (granted, error) in
```
Comment (carmenlau) [inline Swift Chat Demo 2/LoginRegistrationViewController.swift:54]: I think we don't need `center.requestAuthorization` if we will always `registerForRemoteNotifications` again?

File: Swift Chat Demo 2/Swift Chat Demo 2.entitlements [hunk 2]
```diff
@@ -0,0 +1,8 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
+<plist version="1.0">
+<dict>
+	<key>aps-environment</key>
+	<string>development</string>
```
Comment (carmenlau) [inline Swift Chat Demo 2/Swift Chat Demo 2.entitlements:6]: I think we will need an production version for release. We usually create 2 entitlements, `development` for `Debug` and `production` for release.

File: Swift Chat Demo 2.xcodeproj/project.pbxproj [hunk 3]
```diff
@@ -458,12 +470,17 @@
 			baseConfigurationReference = 26D03EB7DD24CC71FAF6D1C6 /* Pods-Swift Chat Demo 2.release.xcconfig */;
 			buildSettings = {
 				ASSETCATALOG_COMPILER_APPICON_NAME = AppIcon;
-				CODE_SIGN_STYLE = Automatic;
-				DEVELOPMENT_TEAM = "";
+				CODE_SIGN_ENTITLEMENTS = "Swift Chat Demo 2/Swift Chat Demo 2.entitlements";
+				"CODE_SIGN_IDENTITY[sdk=iphoneos*]" = "iPhone Distribution";
+				CODE_SIGN_STYLE = Manual;
```
Comment (carmenlau) [inline Swift Chat Demo 2.xcodeproj/project.pbxproj]: After we test it and it works. Please revert the provisioning selection to Automatic, as developers may have their own cert. So we don't have to change it when updating the cert and developers can also use their own certs.

**Review by carmenlau — CHANGES_REQUESTED** [review]

File: Swift Chat Demo 2/Swift Chat Demo 2.entitlements (hunk 2 above)
Comment (howawong) [reply Swift Chat Demo 2/Swift Chat Demo 2.entitlements:6]: I think `requestAuthorization`  has to be called before `registerForRemoteNotifications` ?

Comment (carmenlau) [reply Swift Chat Demo 2/Swift Chat Demo 2.entitlements:6]: Yes, I checked the doc again. If we don't use `requestAuthorization`, the notification will become silent. So you are right! thanks @howawong

https://developer.apple.com/documentation/uikit/uiapplication/1623078-registerforremotenotifications

**Review by carmenlau — APPROVED** [review]

**Pull request merged**